pkg/synthetictests/operators.go

```diff
@@ -229,6 +230,36 @@ func testOperatorOSUpdateStartedEventRecorded(events monitorapi.Intervals, clien
 	return []*junitapi.JUnitTestCase{success}
 }
 
+// testBackoffPullingRegistryRedhatImage looks for this symptom:
+//   reason/ContainerWait ... Back-off pulling image "registry.redhat.io/openshift4/ose-oauth-proxy:latest"
+//   reason/BackOff Back-off pulling image "registry.redhat.io/openshift4/ose-oauth-proxy:latest"
+// and marks it as a flake so we can see how often it's happening.
+//
+func testBackoffPullingRegistryRedhatImage(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+
+	image_name := "\"registry.redhat.io/openshift4"
```

Comment:
any image from registry.redhat.io

Reply:
in what context are we going to be running this test/job (against what payload) that we'd expect registry.redhat.io pullspecs?

i'd expect either `registry.ci.openshift.org`(image from ci payload), `quay.io`(image from nightly payload) or `image-registry.openshift-image-registry.svc:5000`(resolved via the oauth-proxy imagestream w/ pullthrough)

if we have components/test code that's setting up pods to use the registry.redhat.io image directly, we might need to be asking why.

Reply:
we have found image pull from registry.redhat.io to be unreliable.  It happens enough that it's noticeable so we'd like to track it.  The mitigation is usually to rerun the job where it happened.  In a recent case, the problem happened when trying to bring up loki during an [e2e-gcp-upgrade job in CI](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26854/pull-ci-openshift-origin-master-e2e-gcp-upgrade/1496240296596344832).

This PR introduces a test (as a flake) so it won't affect pass/fail numbers.